app/models/rating.rb

```diff
@@ -1,8 +1,19 @@
 # frozen_string_literal: true
 
 class Rating < ApplicationRecord
-  validates :rating, numericality: {only_integer: true, less_than_or_equal_to: 5, greater_than: 0}
+  SUBCATEGORIES = %i[originality description_quality test_quality model_solution_quality].freeze
+  CATEGORIES = SUBCATEGORIES + %i[overall_rating]
+
+  validates :overall_rating, numericality: {only_integer: true, less_than_or_equal_to: 5, greater_than: 0}
+  validates :originality, numericality: {only_integer: true, less_than_or_equal_to: 5, greater_than: 0}
+  validates :description_quality, numericality: {only_integer: true, less_than_or_equal_to: 5, greater_than: 0}
+  validates :test_quality, numericality: {only_integer: true, less_than_or_equal_to: 5, greater_than: 0}
+  validates :model_solution_quality, numericality: {only_integer: true, less_than_or_equal_to: 5, greater_than: 0}
```

Comment:
I am not sure whether I like the meta programming more, but an alternative would be:

```suggestion
CATEGORIES.each do |category|
  validates category, numericality: {only_integer: true, less_than_or_equal_to: 5, greater_than: 0}
end
```

Reply:
I think that is better (also because it reduces the amount of code that needs to be changed if these categories ever change).